cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTableAdmin.java

```diff
@@ -26,6 +30,24 @@
    * Create a StructuredTable using the {@link StructuredTableSpecification}.
    *
    * @param spec table specification
+   * @throws IOException if there is an error creating the table
+   */
+  void create(StructuredTableSpecification spec) throws IOException;
+
+  /**
+   * Get the {@link StructuredTableSpecification} corresponding to the given table id
+   *
+   * @param tableId the table id
+   * @return the specification for the table
+   */
+  @Nullable
+  StructuredTableSpecification getSpecification(StructuredTableId tableId);
+
+  /**
+   * Drop the StructuredTable
+   *
+   * @param tableId the table id
+   * @throws IOException if there is an error dropping the table
    */
-  void create(StructuredTableSpecification spec);
+  void drop(StructuredTableId tableId) throws IOException;
```

Comment:
What if the tableId doesn't exist? Will it throw exception or is an no-op?

Reply:
It the table does not exist, it will be no-op. Added to javadoc